cmd/build/v2/build.go

```diff
@@ -65,6 +65,7 @@ type oktetoBuilderConfigInterface interface {
 	IsOkteto() bool
 	GetAnonymizedRepo() string
 	GetBuildContextHash(*model.BuildInfo) string
+	IsSmartBuildsEnable() bool
```

Comment:
I don't like adding this in directly on the main struct. Following best practices for interfaces we should be creating another interface for checking the `IsSmartBuildsEnable` and adding it as a composition for the main struct. This way we can create a simple struct for the tests that checks only `IsSmartBuildsEnable` without the need of implementing the rest of the functions

Reply:
I would keep it that way, I see no reason why this should be interface composition and not the other methods. Maybe there is something that escapes me but I would try to keep consistency.